src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -284,32 +288,29 @@ public bool ValidateBody(ReadOnlySpan<byte> container, in EofHeader header)
                 return false;
             }
 
-            ReadOnlySpan<byte> typesection = container.Slice(typeSectionStart, typeSectionSize);
-            if (!ValidateTypeSection(typesection))
+            ReadOnlyMemory<byte> typesection = container.Slice(typeSectionStart, typeSectionSize);
+            if (!ValidateTypeSection(typesection.Span))
             {
                 if (Logger.IsTrace) Logger.Trace($"EIP-4750: invalid typesection found");
                 return false;
             }
 
-            for (int sectionIdx = 0; sectionIdx < header.CodeSections.Length; sectionIdx++)
+            int validSection = 0;
+            Parallel.For(0, header.CodeSections.Length, (sectionIdx, state) =>
             {
                 SectionHeader sectionHeader = header.CodeSections[sectionIdx];
                 (int codeSectionStartOffset, int codeSectionSize) = sectionHeader;
-                ReadOnlySpan<byte> code = container.Slice(codeSectionStartOffset, codeSectionSize);
-                if (!ValidateInstructions(code, header))
+                ReadOnlySpan<byte> code = container.Span.Slice(codeSectionStartOffset, codeSectionSize);
+                if (!ValidateInstructions(code, header) ||
+                    !ValidateStackState(sectionIdx, code, typesection.Span, header))
                 {
-                    if (Logger.IsTrace) Logger.Trace($"EIP-3670 : CodeSection {sectionIdx} contains invalid body");
-                    return false;
-                }
-
-                if (!ValidateStackState(sectionIdx, code, typesection, header))
-                {
-                    if (Logger.IsTrace) Logger.Trace($"EIP-5450 : CodeSection {sectionIdx} create has Invalid stack configuration");
-                    return false;
+                    state.Stop();
```

Comment:
Replace `validSection` with: `bool validSections = true` and set it to `false` here. Will avoid `Interlocked.Increment` which needs thread synchronization.